libr/bin/format/pe/pe.c

```diff
@@ -574,12 +574,113 @@ static int bin_pe_parse_imports(struct PE_(r_bin_pe_obj_t)* bin,
 	return false;
 }
 
+#if R_SYS_ENDIAN
+static void PE_(image_dos_header_swapendian)(PE_(image_dos_header) *header) {
+#define swapendian(field) r_mem_swapendian ((ut8*) &header->field, (ut8*) &header->field, sizeof (header->field))
```

Comment:
please use r_read APIs, which take care of this depending on R_SYS_ENDIAN

Reply:
I guess you mean r_buf_read_lexx since we are dealing with RBuffer instead of ut8 array.